### PR TITLE
Proper multiple item images handling

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
Fixes WEBUMENIA-703

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](https://github.com/SlovakNationalGallery/web-umenia-2/blob/dea47a352d25bbc406f7fcaed036d872cdced1da/CHANGELOG.md)
- [x] I have assigned someone to this PR
- [ ] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: app/Http/Controllers/AuthorityController.php [hunk 1]
```diff
@@ -86,18 +86,6 @@ public function store()
...
-    /**
-     * Display the specified resource.
-     *
-     * @param  int  $id
-     * @return Response
-     */
-    public function show($id)
-    {
-        $authority = Authority::find($id);
-        return view('authorities.show')->with('authority', $authority);
```
Comment (rastislav-chynoransky) [inline app/Http/Controllers/AuthorityController.php]: This got no corresponding template

File: app/Http/Controllers/SpiceHarvesterController.php [hunk 2]
```diff
@@ -901,25 +906,29 @@ private function parseDate($string)
 
     private function parseYear($string)
     {
-        return (int)end((explode('.', $string)));
+        $exploded = explode('.', $string);
+        return (int)end($exploded);
     }
 
-    private function resolveIIPUrl($iip_resolver)
+    private function resolveIIPUrls($iip_resolver)
```
Comment (rastislav-chynoransky) [inline app/Http/Controllers/SpiceHarvesterController.php]: Is it ok to create more than one image per item? Are the data checked?

File: resources/views/authorities/show.blade.php [hunk 3]
```diff
@@ -1,123 +0,0 @@
-<!DOCTYPE html>
```
Comment (rastislav-chynoransky) [inline resources/views/authorities/show.blade.php]: Duplicate of `resources/views/items/show.blade.php` (not used)

Comment (igor-kamil) [reply resources/views/authorities/show.blade.php]: good point. can be updated to display authority detail (would be nice). or just remove it.

File: app/Http/Controllers/SpiceHarvesterController.php (hunk 2 above)
Comment (igor-kamil) [reply app/Http/Controllers/SpiceHarvesterController.php]: yes. sure!

File: app/Http/Controllers/AuthorityController.php (hunk 1 above)
Comment (igor-kamil) [reply app/Http/Controllers/AuthorityController.php]: as I wrote below - you can either update `resources/views/authorities/show.blade.php` or remove that.
I would prefer the first option - can be done also just partially (not necessary to display everything and all relationships)

Comment (rastislav-chynoransky) [reply app/Http/Controllers/AuthorityController.php]: I'd rather do it in different PR

File: app/Http/routes.php [hunk 4]
```diff
@@ -184,15 +184,9 @@ function()
 
         $item = Item::find($id);
 
-        if (empty($item) || !$item->isFreeDownload()) {
+        if (empty($item) || !$this->publicDownload()) {
```
Comment (igor-kamil) [inline app/Http/routes.php]: should be `$item->publicDownload()` instead of `$this->publicDownload()`

File: app/Item.php [hunk 5]
```diff
@@ -644,11 +635,6 @@ private function relatedAuthorityIds()
         return $ids;
     }
 
-    public function isFreeDownload()
```
Comment (igor-kamil) [inline app/Item.php]: why to remove this method? you need to check this condition several times all around the code - now you have to use combination of `($item->isFree() && !$item->getZoomableImages()->isEmpty())` or `($item->isFree() && $item->has_iip)`
also there is a space to override the conditions for downloading the hi-res  image. what might be useful for other instances (e.g. to put in action custom "_rights_" etc...)

File: app/Http/routes.php (hunk 4 above)
Comment (rastislav-chynoransky) [reply app/Http/routes.php]: Thanks!

**Pull request closed without merging**